### ISSUE TITLE
ContractFunctionParams().addStringArray(stringArray) returns CONTRACT_REVERT_EXECUTED status in Kotlin-Android

### ISSUE THREAD

**Issue by mohit26chauhan** [hashgraph/hedera-sdk-java#376]
I'm using Java-SDK in Android (Kotlin). I'm using addStringArray() method of ContractFunctionParams() class to send array of strings but it gives me CONTRACT_REVERT_EXECUTED status. I'm creating string array in Kotlin like this: 
`val paramArray = arrayOf("value1","value2","value3,...")`

Then calling Contract method like this:
`val functionParams = ContractFunctionParams().addStringArray(paramArray)`

`val record = ContractExecuteTransaction()`
`        .setMaxTransactionFee(Hbar(555))`
`                .setContractId(contractId)`
`                .setGas(16050)`
`                .setFunction("contractMethodName", functionParams)`
`                .execute(client)`
`                .getRecord(client)`

Above code throws `HederaStatusException` exception with status `CONTRACT_REVERT_EXECUTED`

**But in Node JS, it is working fine.** Node JS code is attached and the values format that I need to send is circled. I want the equivalent way in Kotlin as I'm doing in Node JS. Please help!

![JavaScript](https://user-images.githubusercontent.com/6298935/76696851-ea0c2680-66b5-11ea-95ec-8b5c6d82ca13.jpeg)

**Comment by mehcode** [comment]
Did you resolve this? What was the issue (for posterity)?

**Comment by mohit26chauhan** [comment]
We didn't found any solution. On contract server side, we change out implementation to accept multiple string values in the method instead of one single string array. In this situation, we got compilation error similar to this - "Number of parameters exceeds for a function" because maximum 10 parameters are supported in contract method, I guess. So, for now we are sending the extra data in the form of concatenated string. This is just a hack.

**Comment by abonander** [comment]
I did find the issue in our Solidity encoding of `string[]`; we don't correctly calculate the dynamic offsets of the individual strings inside the array (I originally assumed it was from the start of the array but it's actually from the start of the encoded arguments as a whole). It'll take a bit of internal refactoring to fix this, might take me a day or two.

**Comment by abonander** [comment]
Turns out my initial assumption _was_ correct, but there must have been a different bug in the existing implementation. Solidity's specifications are pretty dense, though; it took another read through of the example encoding [here](https://solidity.readthedocs.io/en/v0.6.4/abi-spec.html#use-of-dynamic-types) to realize this.

**Comment by abonander** [comment]
I have a fix in #381 if you want to try it locally:

```bash
$ git clone https://github.com/hashgraph/hedera-sdk-java
$ cd hedera-sdk-java
$ git checkout ab/solidity-string-array
$ mvn install
```

Then run your code again.

**Comment by abonander** [comment]
#381 has been merged and branch `ab/solidity-string-array` has been deleted so if you still want to try it don't bother with the `git checkout` part and just run `mvn install` for master.
